### PR TITLE
LSTM Suport

### PR DESCRIPTION
**Pull request by psyyz10**
## What changes were proposed in this pull request?

Add LSTM Suport

## How was this patch tested?
Unit tests

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/415

### CODE REVIEW

File: dl/src/main/scala/com/intel/analytics/bigdl/nn/LSTM.scala [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+ * D. https://github.com/wojzaremba/lstm
+ *
+ * @param inputSize the size of each input vector
+ * @param hiddenSize Hidden unit size in the LSTM
+ * @param p p is the threshold for Dropout. If p equals 0, create
+ *          a LSTM without Dropout
+ */
+class LSTM[T : ClassTag] (
+  inputSize: Int = 4,
+  hiddenSize: Int = 3,
```
Comment (jason-dai) [inline dl/src/main/scala/com/intel/analytics/bigdl/nn/LSTM.scala]: What's the point of giving inputSize and hiddenSize default values?

**Comment by jason-dai** [comment]
We need reset, clearState, equals, hashCode, toString, etc., yes?

File: dl/src/main/scala/com/intel/analytics/bigdl/nn/LSTM.scala [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+          .add(Dropout(p))
+          .add(Dropout(p)))
+        .add(ParallelTable()
+          .add(Linear(inputSize, hiddenSize))
+          .add(Linear(inputSize, hiddenSize))
+          .add(Linear(inputSize, hiddenSize))
+          .add(Linear(inputSize, hiddenSize)))
+        .add(JoinTable(2, 1))
+
+      val o2g = Sequential()
```
Comment (jason-dai) [inline dl/src/main/scala/com/intel/analytics/bigdl/nn/LSTM.scala]: where is o2g used?

**Comment by psyyz10** [comment]
Done

**Comment by jason-dai** [comment]
LGTM

**Pull request merged**